seip2023_practical_assignments/unittesting/src/test/java/math/ArithmeticOperationsTest.java

```diff
@@ -0,0 +1,258 @@
+     */
+    @Test
+    public void test_Divide_NormalCase2() {
+        double numerator = -6;
+        double denominator = 3;
+        double expected = -2.0;
+
+        double result = arithmeticOperations.divide(numerator, denominator);
+
+        assertEquals(expected, result, 0.00001);
+    }
+
+    /**
+     * Test case for dividing zero by a positive number.
+     */
+    @Test
+    public void test_Divide_NormalCase3() {
+        double numerator = 0;
+        double denominator = 5;
+        double expected = 0.0;
+
+        double result = arithmeticOperations.divide(numerator, denominator);
+
+        assertEquals(expected, result, 0.00001);
+    }
+
+    /**
+     * Test case for dividing by zero.
+     * Expects an ArithmeticException to be thrown.
+     */
+    @Test(expected = ArithmeticException.class)
+    public void test_Divide_DivisionByZero() {
+        double numerator = 0;
+        double denominator = 0;
+
+        arithmeticOperations.divide(numerator, denominator);
+    }
+
+    /**
+     * Test case for dividing a large number by 1.
+     */
+    @Test
+    public void test_Divide_LargeNumbers() {
+        double numerator = Double.MAX_VALUE;
+        double denominator = 1;
+        double expected = 1.7976931348623157E308;
+
+        double result = arithmeticOperations.divide(numerator, denominator);
+
+        assertEquals(expected, result, 0.00001);
+    }
+
+    /**
+     * Test case for dividing a negative number by zero.
+     * Expects an ArithmeticException to be thrown.
+     */
+    @Test(expected = ArithmeticException.class)
+    public void testDivide_NegativeNumberDividedByZero() {
+        double numerator = -10;
+        double denominator = 0;
+
+        arithmeticOperations.divide(numerator, denominator);
+    }
+
+    /**
+     * Test case for dividing the maximum value of a double by the maximum value of a double.
+     */
+    @Test
+    public void test_Divide_DivideByMaxValue() {
+        double numerator = Double.MAX_VALUE;
+        double denominator = Double.MAX_VALUE;
+        double expected = 1.0;
+
+        double result = arithmeticOperations.divide(numerator, denominator);
+
+        assertEquals(expected, result, 0.00001);
+    }
+
+    @Rule 
+	public ExpectedException thrown = ExpectedException.none(); //initialize it to .none()
+    
+    ArithmeticOperations operations = new ArithmeticOperations();
+
+    /**
+     * Test case for multiplying two positive numbers.
+     */
+    @Test
+    public void test_Multiply_NormalCase1() {
+        int x = 5;
+        int y = 3;
+        int expected = 15;
+
+        int result = operations.multiply(x, y);
+
+        assertEquals(expected, result);
+    }
+
+    /**
+     * Test case for multiplying a number by zero.
+     */
+    @Test(expected = ArithmeticException.class)
+    public void test_Multiply_NormalCase2() {
+        int x = 10;
+        int y = 0;
+
+        operations.multiply(x, y);
+    }
```

Comment:
You should allow multiplications with `y=0` like you do with `x=0` later in your tests.